### DISCUSSION TITLE
PyOptiSLang logging

### DISCUSSION THREAD

**Discussion by ansDArnold** [ansys/pyoptislang#31] [General]
Documentation optiSLang logging (doc/api/logging.html) advises to use a protected member funtion `Optislang._logger()` which is not very pythonic

Wouldn't it be better to have a function like
```
osl.log(message, level)
osl.log.debug(message)
osl.log_debug(message)
```

What do you think?

**Comment by vymlatilp** [comment]
Hi Daniel,

there are 2 options to fix it:
1/ the normal user will not use it as it is protected attribute and we will not document it. 
2/ we make it public and update documentation

However as you can see the [logging documentation of PyMAPDL](https://mapdl.docs.pyansys.com/contribution_and_api/logging.html), they did it in the same way. 
We will discuss with @rfahlberg how to handle this.

Best regards,

Petr

**Reply by ansDArnold** [reply]
Hi Petr,

I, personally, like option 2.

Are there other opinions, anybody?

**Reply by rfahlberg** [reply]
@vymlatilp @danielarnold0455 I agree.
